### PR TITLE
Bump go.mod k8s.io to 1.26

### PR DESCRIPTION
**Pull request by fmuyassarov**
Bump go.mod k8s.io to 1.26 and other dependencies upgrade.

### CODE REVIEW

**Comment by marquiz** [comment]
Thanks @fmuyassarov for the update!

It wasn't that easy, though 😅 E2e-tests cannot run because of some misalignment with the updated k8s e2e-test-framework

```
test/e2e/utils/pod/pod.go:114:4: f.PodClient undefined (type *framework.Framework has no field or method PodClient)
test/e2e/utils/pod/pod.go:114:55: undefined: framework.DefaultPodDeletionTimeout
```

**Comment by fmuyassarov** [comment]
> Thanks @fmuyassarov for the update!
> 
> It wasn't that easy, though sweat_smile E2e-tests cannot run because of some misalignment with the updated k8s e2e-test-framework
> 
> ```
> test/e2e/utils/pod/pod.go:114:4: f.PodClient undefined (type *framework.Framework has no field or method PodClient)
> test/e2e/utils/pod/pod.go:114:55: undefined: framework.DefaultPodDeletionTimeout
> ```

Yup, I just got them fixed. There are quite some updates. Will push the changes in few mins

**Comment by fmuyassarov** [comment]
I didn't notice those issues until I submitted the PR. Because usually I run make test, while those issues were caught by golanci-lint.

**Comment by marquiz** [comment]
Still got some error
```
I1212 23:05:46.389382    1734 e2e.go:126] Starting e2e run "1d04aa90-691c-4ddd-b16e-41da0d23ca2c" on Ginkgo node 1
Invalid Function
ginkgo.AfterEach("Node Feature Discovery topology updater CRD and RBAC removal", func(f *framework.Framework, failed bool) {
/home/marquiz/go/git/node-feature-discovery/review/test/e2e/topology_updater.go:103
  [AfterEach] node must be passed func(), func(ctx SpecContext), or func(ctx
  context.Context).
  You passed func(*framework.Framework, bool) instead.

  Learn more at: http://onsi.github.io/ginkgo/#node-decorators-overview
```

**Comment by fmuyassarov** [comment]
Thanks for catching those e2e issues @marquiz . PTAL.

File: go.mod [hunk 1]
```diff
@@ -3,34 +3,34 @@ module sigs.k8s.io/node-feature-discovery
...
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/protobuf v1.5.2
-	github.com/google/go-cmp v0.5.8
+	github.com/google/go-cmp v0.5.9
 	github.com/jaypipes/ghw v0.8.1-0.20210827132705-c7224150a17e
 	github.com/k8stopologyawareschedwg/noderesourcetopology-api v0.0.12
-	github.com/klauspost/cpuid/v2 v2.2.2
-	github.com/onsi/ginkgo/v2 v2.3.1
-	github.com/onsi/gomega v1.22.1
+	github.com/klauspost/cpuid/v2 v2.1.0
```
Comment (marquiz) [inline go.mod:13]: Please don't downgrade this

Comment (fmuyassarov) [reply go.mod:13]: that's weird, I was not planning 😅 not sure how it happened

Comment (fmuyassarov) [reply go.mod]: fixed

**Review by marquiz — APPROVED** [review]
Looks good now
/lgtm

**Pull request merged**